README.md

```diff
@@ -18,27 +18,32 @@ Based on [redux-saga-rxjs](https://github.com/salsita/redux-saga-rxjs) by Salsit
 npm install store-saga --save
 ```
 
-Write a saga:
+Write a saga (you _don't_ have to import and specify all those types, but they help to spot errors earlier):
+
 ```ts
+import {Observable} from 'rxjs/Observable';
+import {Action} from '@ngrx/store';
 import {createSaga} from 'store-saga';
 
-export const increment = createSaga(function(){
-  return iteration$ => iteration$
-    .filter(iter => iter.action.type === 'DECREMENT')
-    .map(() => {
-      return { type: 'INCREMENT'}
-    });
+export const incrementProvider = createSaga(function() {
+  return function incrementSaga(iteration$): Observable<Action> {
+    return iteration$
+      .filter(iter => iter.action.type === 'DECREMENT')
+      .map(() => {
```

Comment:
We're having issues with all RxJs operators in VS 2015, although they compile fine (even without the generic type added): 

![map-squiggle](https://cloud.githubusercontent.com/assets/3185573/14564466/f15606a0-0326-11e6-8998-deb4ec8ea6da.png)

so I can just follow your suggestion, "blindly", so to say.